### PR TITLE
Upgrade dependencies (latest)

### PR DESCRIPTION
**Pull request by qligier**
Upgrade dependencies:

- HAPI to 6.4.4
- FHIR Core to 5.6.117
- Spring to 5.3.27
- Spring Boot to 2.7.11

### CODE REVIEW

**Comment by qligier** [comment]
Test failure: maps

The two tests failing are `FhirMappingLanguageTests.testTutorialStep1()` and `testTutorialStep2()`. The error is `org.hl7.fhir.r5.utils.FHIRLexer$FHIRLexerException: Error @5, 1: Found "uses" expecting "map"`.

The map is an R5 map:
```
/// url = 'http://hl7.org/fhir/StructureMap/tutorial-step1'
/// name = 'tutorial-step1'
/// title = 'Tutorial Step 1'

uses "http://hl7.org/fhir/StructureDefinition/tutorial-left1" alias TLeft1 as source
uses "http://hl7.org/fhir/StructureDefinition/tutorial-right1" alias TRight1 as target

group tutorial(source src : TLeft1, target tgt : TRight1) {
    src.a as a -> tgt.a = a "rule_a";
}
```

while the lexer awaits an R4 map (starting with `map`). The engine is R4 (`new MatchboxEngineBuilder().getEngineR4()`).

**Comment by oliveregger** [comment]
that should be an R5 map in my view, event with getEngineR4

File: matchbox-engine/src/main/java/org/hl7/fhir/r5/context/BaseWorkerContext.java [hunk 1]
```diff
@@ -136,2285 +136,2267 @@ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWIS
...
+	private Set<String> notCanonical = new HashSet<String>();
+
+	protected IWorkerContextManager.IPackageLoadingTracker packageTracker;
+
+	@Override
+	public Resource fetchResourceById(String type, String uri) {
+		synchronized (lock) {
+			String[] parts = uri.split("\\/");
+			if (!Utilities.noString(type) && parts.length == 1) {
+				if (allResourcesById.containsKey(type)) {
```
Comment (oliveregger) [inline matchbox-engine/src/main/java/org/hl7/fhir/r5/context/BaseWorkerContext.java]: if (!Utilities.noString(type) && parts.length == 1) {
        if (allResourcesById.containsKey(type)) {
          // matchbox prevent NP
          ResourceProxy proxy = allResourcesById.get(type).get(parts[0]);
          return proxy != null ?  proxy.getResource() : null;
        } else {
          return null;
        }

File: pom.xml [hunk 2]
```diff
@@ -39,8 +39,8 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <java.version>11</java.version>
-        <fhir.core.version>5.6.116</fhir.core.version>
-        <hapi.fhir.version>6.4.1</hapi.fhir.version>
+        <fhir.core.version>5.6.971</fhir.core.version>
```
Comment (oliveregger) [inline pom.xml]: accoding to https://github.com/hapifhir/org.hl7.fhir.core/releases  [5.6.117](https://github.com/hapifhir/org.hl7.fhir.core/releases/tag/5.6.117) is the latest release

Comment (qligier) [reply pom.xml]: Unfortunately, HAPI 6.4.4 depends on FHIR core 5.6.971: https://github.com/hapifhir/hapi-fhir/blob/v6.4.4/pom.xml#L867

File: matchbox-engine/src/main/java/org/hl7/fhir/r5/context/BaseWorkerContext.java (hunk 1 above)
Comment (qligier) [reply matchbox-engine/src/main/java/org/hl7/fhir/r5/context/BaseWorkerContext.java]: Thanks, fixed in 8dad8bdb87d330fe202cef3c9f8de1264d029cd2

**Comment by qligier** [comment]
Refactored as discussed. No update needed for the HAPI core classes.

**Comment by qligier** [comment]
Closed in favor of #107.

**Pull request closed without merging**